tests/test_equipment.py

```diff
@@ -0,0 +1,414 @@
+import os
+from pathlib import Path
+from typing import Dict
+
+import numpy as np
+import pandas as pd
+import pytest
+
+from tlo import Date, Module, Simulation
+from tlo.analysis.utils import parse_log_file
+from tlo.events import IndividualScopeEventMixin
+from tlo.methods import Metadata, demography, healthsystem
+from tlo.methods.equipment import Equipment
+from tlo.methods.hsi_event import HSI_Event
+
+resourcefilepath = Path(os.path.dirname(__file__)) / "../resources"
+
+
+def test_core_functionality_of_equipment_class(seed):
+    """Test that the core functionality of the equipment class works on toy data."""
+
+    # Create toy data
+    catalogue = pd.DataFrame(
+        [
+            {"Item_Description": "ItemZero", "Item_Code": 0, "Pkg_Name": float('nan')},
+            {"Item_Description": "ItemOne", "Item_Code": 1, "Pkg_Name": float('nan')},
+            {"Item_Description": "ItemTwo", "Item_Code": 2, "Pkg_Name": 'PkgWith2+3'},
+            {"Item_Description": "ItemThree", "Item_Code": 3, "Pkg_Name": 'PkgWith2+3'},
+        ]
+    )
+    data_availability = pd.DataFrame(
+        # item 0 is not available anywhere; item 1 is available everywhere; item 2 is available only at facility_id=1
+        # No data for fac_id=2
+        [
+            {"Item_Code": 0, "Facility_ID": 0, "Pr_Available": 0.0},
+            {"Item_Code": 0, "Facility_ID": 1, "Pr_Available": 0.0},
+            {"Item_Code": 1, "Facility_ID": 0, "Pr_Available": 1.0},
+            {"Item_Code": 1, "Facility_ID": 1, "Pr_Available": 1.0},
+            {"Item_Code": 2, "Facility_ID": 0, "Pr_Available": 0.0},
+            {"Item_Code": 2, "Facility_ID": 1, "Pr_Available": 1.0},
+        ]
+    )
+
+    mfl = pd.DataFrame(
+        [
+            {'District': 'D0', 'Facility_Level': '1a', 'Region': 'R0', 'Facility_ID': 0, 'Facility_Name': 'Fac0'},
+            {'District': 'D0', 'Facility_Level': '1b', 'Region': 'R0', 'Facility_ID': 1, 'Facility_Name': 'Fac1'},
+            {'District': 'D0', 'Facility_Level': '2', 'Region': 'R0', 'Facility_ID': 2, 'Facility_Name': 'Fac2'},
+        ]
+    )
+
+    # Create instance of the Equipment class with these toy data and check availability of equipment...
+    # -- when using `default` behaviour:
+    eq_default = Equipment(
+        catalogue=catalogue,
+        data_availability=data_availability,
+        rng=np.random.RandomState(np.random.MT19937(np.random.SeedSequence(seed))),
+        master_facilities_list=mfl,
+        availability="default",
+    )
+
+    # Checks on parsing equipment items
+    # - using single integer for one item_code
+    assert {1} == eq_default.parse_items(1)
+    # - using list of integers for item_codes
+    assert {1, 2} == eq_default.parse_items([1, 2])
+    # - using single string for one item descriptor
+    assert eq_default.parse_items('ItemOne')
+    # - using list of strings for item descriptors
+    assert eq_default.parse_items(['ItemOne', 'ItemTwo'])
```

Comment:
The assertions here seem to be just that the values returned by `parse_items` are not 'Falsey' (for example not an empty set). Could we not instead check that we get the expected set of item codes (that is I think `{1}` and `{1, 2}` respectively if I'm understanding how `parse_items` works correctly)?